Description: Don't try to announce the finish of a NULL job.
Origin: vendor, Ubuntu
Bug-Ubuntu: LP #345754, #838464

--- a/daemon/gvfschannel.c
+++ b/daemon/gvfschannel.c
@@ -611,7 +611,8 @@
 
   job = channel->priv->current_job;
   channel->priv->current_job = NULL;
-  g_vfs_job_emit_finished (job);
+  if (job)
+    g_vfs_job_emit_finished (job);
 
   class = G_VFS_CHANNEL_GET_CLASS (channel);
   
@@ -632,7 +633,7 @@
     }
   /* Start queued request or readahead */
   else if (!start_queued_request (channel) &&
-	   class->readahead)
+	   class->readahead && job)
     {
       /* No queued requests, maybe we want to do a readahead call */
       channel->priv->current_job = class->readahead (channel, job);
